src/imu/imu.c

```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
 		int zeroValue = g_imu_filter[physicalChannel].current_value;
 		//adjust for gravity
 		float countsPerUnit = imu_device_counts_per_unit(physicalChannel);
-		if (c->physicalChannel == IMU_CHANNEL_Z) zeroValue-= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
+		if (i == IMU_CHANNEL_Z) zeroValue -= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
```

Comment:
Case dependant methinks.  I really dislike them as a matter of preference (Kernel standard states "Don't put multiple statements on a single line unless you have something to hide"), but as with most things in C, they have their place if used sparingly.